tests/src/FunctionalJavascript/CollaboraModalFormatterTest.php

```diff
@@ -0,0 +1,315 @@
+    $button = $this->assertPreviewButton($media);
+
+    // No dialog or iframe exists (yet).
+    $assert_session->elementNotExists('css', '.ui-dialog');
+    $assert_session->elementNotExists('css', '.ui-dialog-content');
+    $assert_session->elementNotExists('css', 'iframe');
+
+    // Clicking the button opens the modal.
+    $button->click();
+    $iframe = $this->assertModalWithIframe($media, 750);
+
+    // Clicking the close button closes the modal.
+    $assert_session->elementExists('named', ['button', 'Close'], $iframe->getParent()->getParent())->click();
+    // All parts of the dialog are gone.
+    $assert_session->elementNotExists('css', '.ui-dialog');
+    $assert_session->elementNotExists('css', '.ui-dialog-content');
+    $assert_session->elementNotExists('css', 'iframe');
+
+    // Test with different themes.
+    $this->setActiveTheme('claro');
+    $this->drupalGet($media->toUrl());
+    $this->assertPreviewButton($media)->click();
+    $this->assertModalWithIframe($media, 750);
+
+    $this->setActiveTheme('olivero');
+    $this->drupalGet($media->toUrl());
+    $this->assertPreviewButton($media)->click();
+    $this->assertModalWithIframe($media, 750);
+
+    // Test narrow viewport in different themes.
+    $this->getSession()->resizeWindow(400, 800);
+    $this->drupalGet($media->toUrl());
+    $this->assertPreviewButton($media)->click();
+    $this->assertModalWithIframe($media);
+
+    $this->setActiveTheme('claro');
+    $this->drupalGet($media->toUrl());
+    $this->assertPreviewButton($media)->click();
+    $this->assertModalWithIframe($media);
+
+    $this->setActiveTheme('stark');
+    $this->drupalGet($media->toUrl());
+    $this->assertPreviewButton($media)->click();
+    $this->assertModalWithIframe($media);
+  }
+
+  /**
+   * Asserts that a preview button exists as expected, and returns it.
+   *
+   * @param \Drupal\media\MediaInterface $media
+   *   The media entity being displayed.
+   *
+   * @return \Behat\Mink\Element\NodeElement
+   *   The preview button.
+   */
+  protected function assertPreviewButton(MediaInterface $media): NodeElement {
+    $assert_session = $this->assertSession();
+    $button = $assert_session->elementExists('named', ['link', 'Preview']);
+
+    // Assert button attributes.
+    $this->assertSame('/cool/modal/' . $media->id(), $button->getAttribute('href'));
+    $this->assertTrue($button->hasClass('use-ajax'));
+    $this->assertTrue($button->hasClass('button'));
+    $this->assertTrue($button->hasClass('button--small'));
+    $this->assertSame('modal', $button->getAttribute('data-dialog-type'));
+    $this->assertSame('{"width":750,"classes":{"ui-dialog":"cool-modal-preview"}}', $button->getAttribute('data-dialog-options'));
+
+    // Wait until the click handler is attached.
+    $this->getSession()->getPage()->waitFor(10, function () use ($button) {
+      return $button->getAttribute('data-once') === 'ajax';
+    });
```

Comment:
Remove, it's wrong.

Reply:
Why is it wrong?

Reply:
Actually we don't need to wait here, it just works. All the "document ready " happens before this click.\
Removing.